src/scripts/parser.js

```diff
@@ -56,12 +65,26 @@ exports.deflate = function (json, prefix) {
  * @returns {{}}
  */
 var inflateItem = function (keys, value, result) {
+    var key = keys[ 0 ];
```

Comment:
While inflating, if the key is a number, then we are inside an array, use the key as an numeric index and create the array if if not yet created.